doc/notes/plutus-core/Builtins_Overview.md

```diff
@@ -0,0 +1,350 @@
+    previous point).
+
+We propose that the induced built-in functions should be
+expressible in Typed Plutus Core[^2] (modulo the effect of logging which
+has a special status). That is, built-in functions should not extend
+the possible semantics. Extending the model by adding a built-in
+function not representable in Plutus could be a potential *security
+concern*, as the interaction of a non-standard semantics with the
+rest of the language is not guaranteed to be safe, or to respect the
+cost model. A new built-in function that extends the possible
+semantics would require rethinking the whole specification,
+formalisation, and cost semantics, so the bar for the addition of
+such a built-in should be set extremely high.
+
+[^1]: For some examples see `plutus-core/examples/PlutusCore/Examples/Builtins.hs`.
+[^2]: Asking for a typed implementation, rather than untyped Plutus Core, gives 
+  further guarantees of good behaviour.
+
+#### Correspondence of Plutus types and Haskell types
+
+Plutus' `integer` corresponds to Haskell's `Integer`. If
+the denotation has in its type `Int`[^3], `Int64`, or
+`Word8`, these types will be automatically converted to/from
+`Integer` after an underflow/overflow check.
+
+[^3]: The type `Int` is convertible only if it’s equal to `Int64` 
+  (32 bit systems are not supported).
+
+### Specification Requirements
+Builtin functions must take at least one argument (either a type or term).
+
+In the specification there is a distinction between variables that
+range over built-in types $v$# (type operators such as `list`
+may be applied to these variables), and variables that range over
+any type $v⋆$ (such as the case of `ifThenElse`). However this
+distinction is not currently made in the implementation.
+
+Let's call *composite built-in types* types obtained from either
+built-in types or saturated built-in type operators applied to
+composite built-in types. All the arguments must be either a
+composite built-in type or a (general) type variable. One
+consequence of this requirement is that no higher-order arguments
+are allowed.
+
+Polymorphism over built-in type operators is not allowed. For
+example, a built-in function of the following type is not allowed:
+```
+forall f. f integer -> bool
+```
+
+The specification allows for universal quantifiers to be
+interspersed with any types, but the scope is always the rest of the
+type, including the return type. If we compare with the case where
+quantification of type variables is the outermost operation, the
+generality of adding quantifiers in the middle does not seem to gain
+any expressivity. Furthermore, it\'s not used in any of the current
+built-ins, for which all polymorphic built-ins have their
+quantifiers at the beginning of the type. Note: Because type
+application gets translated to a `force` operation on `uplc`,
+there could be a difference in expressivity regarding errors.
+However currently, no built-in function uses this corner case.
```

Comment:
Could you elaborate a bit more on this? Don't understand it, sorry.

Reply:
The only place I could think where you would have different semantics  between the quantifiers at the beginning or *interleaved*, is if you get an error in the evaluation of a `force` in the middle of the arguments. This might not longer be true with deferred errors, but I'm not sure.

Reply:
Ah, yes, that's a very good point and thank you for the explanation. It's indeed true that there's difference in expressiveness here, because while we do have deferred unlifting, we don't have deferred interleaving (see [this](https://input-output.atlassian.net/browse/PLT-1353) Jira ticket) and `force` is affected by the latter, not the former.